aws-api/src/main/java/com/amplifyframework/api/aws/utils/RestRequestFactory.java

```diff
@@ -15,78 +15,74 @@
 
 package com.amplifyframework.api.aws.utils;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
 import com.amplifyframework.api.rest.HttpMethod;
 
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.Map;
+import java.util.Objects;
 
 import okhttp3.Headers;
 import okhttp3.HttpUrl;
 import okhttp3.Request;
 import okhttp3.RequestBody;
 
 /**
- * Util class to handle Rest url request.
+ * Factory to create {@link URL}s and OkHttp {@link Request}s for REST-ful behaviors.
  */
-public final class RestOperationRequestUtils {
-
-    private static final String CONTENT_TYPE = "application/json";
-
-    /**
-     * Private constructor since this is a utility class.
-     */
-    private RestOperationRequestUtils() {
-        //not called
-    }
+public final class RestRequestFactory {
+    @SuppressWarnings("checkstyle:all") private RestRequestFactory() {}
 
     /**
      * Create an URL by appending the path and queries.
-     * <p>
      * Throws exception if the url is malformed.
-     *
      * @param endpoint        A valid endpoint
      * @param path            path to be appended.
      * @param queryParameters query parameters
      * @return Valid URL
      * @throws MalformedURLException Throw when the exception is not valid
      */
-    public static URL constructURL(String endpoint,
-                                   String path,
-                                   Map<String, String> queryParameters)
+    @NonNull
+    public static URL createURL(
+            @NonNull String endpoint,
+            @Nullable String path,
+            @Nullable Map<String, String> queryParameters)
             throws MalformedURLException {
-        String urlPath = "";
-        if (path != null) {
-            urlPath = path;
-        }
         URL url = new URL(endpoint);
         HttpUrl.Builder builder = new HttpUrl.Builder()
-                .scheme(url.getProtocol())
-                .host(url.getHost())
-                .addPathSegment(url.getPath().replaceFirst("/", ""))
-                .addPathSegment(urlPath);
+            .scheme(url.getProtocol())
+            .host(url.getHost())
+            .addPathSegment(url.getPath().replaceFirst("/", ""))
```

Comment:
What if the path is "path1/path2" as in your example? Wouldn't this remove that slash?